File: AdventJS/2025/reto 5/5.py
# import these modules if you need them:
from datetime import datetime, timezone
import math
      
def time_until_take_off(from_time: str, take_off_time: str) -> int:
  from_time = from_time[:-3].replace("*", "-").replace("@", " ").replace("|", ":")
  take_off_time = take_off_time[:-3].replace("*", "-").replace("@", " ").replace("|", ":")

  start = datetime.strptime(from_time, "%Y-%m-%d %H:%M:%S")
  end = datetime.strptime(take_off_time, "%Y-%m-%d %H:%M:%S")

  seconds = end - start
  return int(seconds.total_seconds())

File: AdventJS/2025/reto 5/5.js
/**
 * @param {string} fromTime - The current time in elf format
 * @param {string} takeOffTime - The take off time in elf format
 * @returns {number} The time in seconds until take off
 */
function timeUntilTakeOff(fromTime, takeOffTime) {
	fromTime = fromTime.slice(0, -3);
	takeOffTime = takeOffTime.slice(0, -3);

	const diccionario = {
		"*": "-",
		"@": "T",
		"|": ":"
	}
	
	const regex = /\*|@|\|/g;
	
	fromTime = fromTime.replace(regex, (match) => diccionario[match]);
	takeOffTime = takeOffTime.replace(regex, (match) => diccionario[match]);
	
	fromTime += "Z";
	takeOffTime += "Z";
	
	let t1 = new Date(fromTime);
	let t2 = new Date(takeOffTime);
	let resultado = (t2 - t1) / 1000; 
	return resultado;
}